File: POO/poo11048216/Aulas/aula15/Conta/ContaPoupanca.cpp
#ifndef CONTAPOUPANCA_H
#include "ContaPoupanca.h"
#define CONTAPOUPANCA_H
#endif

ContaPoupanca::ContaPoupanca(std::string cpf, std::string nome, int d, int m, int a, float tx):Conta(cpf, nome)
{
    this->setAniversario(Data(d, m, a));
    this->setTxRef(tx);
}
ContaPoupanca::ContaPoupanca(std::string cpf, std::string nome , Data dt, float tx):Conta(cpf, nome)
{
    this->setAniversario(dt);
    this->setTxRef(tx);
}

void ContaPoupanca::setAniversario(Data dt)
{
    this->aniversario = dt;
}

void ContaPoupanca::setTxRef(float tx)
{
    this->txRef = tx;
}

Data ContaPoupanca::getAniversario()
{
    return this->aniversario;
}

float ContaPoupanca::getTxRef()
{
    return this->txRef;
}

void ContaPoupanca::correcao()
{
    if(!this->isBloqueada())
    {
        this->setSaldo(this->getSaldo() * (1.02f));
    }
}

void ContaPoupanca::extrato()
{
    Conta::extrato();
    std::cout << "ANIVERSARIO : " << this->getAniversario().getDia() << std::endl;
    std::cout << "===================================== " << std::endl;
}

File: POO/poo11048216/Aulas/aula15/JConta/ContaPoupanca.java
class ContaPoupanca extends Conta
{
    private Data aniversario;
    private float txRef;
    
    public ContaPoupanca()
    {   }
    
    public ContaPoupanca(String n, String cpf, Data dt)
    {
        super(cpf, n);
        this.setAniversario(dt);
    }
    
    public ContaPoupanca(String n, String cpf, int d, int m, int a)
    {
        super(cpf, n);
        this.setAniversario(new Data(d, m, a));
    }
    
    public void setAniversario(Data d)
    {
        this.aniversario = d;
    }
    
    public void setTxRef(float t)
    {
        this.txRef = t;
    }
    
    public Data getAniversario()
    {
        return this.aniversario;
    }
    
    public float getTxRef()
    {
        return this.txRef;
    }
    
    public void extrato()
    {
        super.extrato();
        System.out.println("ANIVERSARIO .: " + this.getAniversario().getDia());
        System.out.println("===================================");        
    }
}